client/src/main/scala/org/http4s/client/Client.scala

```diff
@@ -220,4 +220,24 @@ final case class Client(open: Service[Request, DisposableResponse], shutdown: Ta
     shutdown.run
 }
 
+object Client {
+  /** Creates a mock client from the specified service.  Useful for generating
+    * pre-determined responses for requests in testing.
+    * 
+    * @param service the service to respond to requests to this client
+    * 
+    * @param dispose action to dispose of responses generated by this client. 
+    * Defaults to a no-op.
+    * 
+    * @param shutdown action when this client is shutdown.  Defaults to a no-op.
+    */
+  def mock(service: HttpService, 
+           dispose: Task[Unit] = Task.now(()),
```

Comment:
For a more realistic experience, this should arguably prevent subsequent reads from the request or response bodies.  Thoughts?